test/framework/src/main/java/org/elasticsearch/AnalysisFactoryTestCase.java

```diff
@@ -0,0 +1,326 @@
+import org.elasticsearch.index.analysis.ShingleTokenFilterFactory;
+import org.elasticsearch.index.analysis.SnowballTokenFilterFactory;
+import org.elasticsearch.index.analysis.SoraniNormalizationFilterFactory;
+import org.elasticsearch.index.analysis.StandardTokenFilterFactory;
+import org.elasticsearch.index.analysis.StandardTokenizerFactory;
+import org.elasticsearch.index.analysis.StemmerOverrideTokenFilterFactory;
+import org.elasticsearch.index.analysis.StemmerTokenFilterFactory;
+import org.elasticsearch.index.analysis.StopTokenFilterFactory;
+import org.elasticsearch.index.analysis.SynonymTokenFilterFactory;
+import org.elasticsearch.index.analysis.ThaiTokenizerFactory;
+import org.elasticsearch.index.analysis.TrimTokenFilterFactory;
+import org.elasticsearch.index.analysis.TruncateTokenFilterFactory;
+import org.elasticsearch.index.analysis.UAX29URLEmailTokenizerFactory;
+import org.elasticsearch.index.analysis.UpperCaseTokenFilterFactory;
+import org.elasticsearch.index.analysis.WhitespaceTokenizerFactory;
+import org.elasticsearch.index.analysis.WordDelimiterTokenFilterFactory;
+import org.elasticsearch.index.analysis.compound.DictionaryCompoundWordTokenFilterFactory;
+import org.elasticsearch.index.analysis.compound.HyphenationCompoundWordTokenFilterFactory;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+/** 
+ * Alerts us if new analyzers are added to lucene, so we don't miss them.
+ * <p>
+ * If we don't want to expose one for a specific reason, just map it to Void.
+ * The deprecated ones can be mapped to Deprecated.class.
+ */
+public class AnalysisFactoryTestCase extends ESTestCase {
+
+    static final Map<String,Class<?>> KNOWN_TOKENIZERS = Collections.unmodifiableMap(new HashMap<String,Class<?>>() {{
+        // exposed in ES
+        put("classic",       ClassicTokenizerFactory.class);
+        put("edgengram",     EdgeNGramTokenizerFactory.class);
+        put("keyword",       KeywordTokenizerFactory.class);
+        put("letter",        LetterTokenizerFactory.class);
+        put("lowercase",     LowerCaseTokenizerFactory.class);
+        put("ngram",         NGramTokenizerFactory.class);
+        put("pathhierarchy", PathHierarchyTokenizerFactory.class);
+        put("pattern",       PatternTokenizerFactory.class);
+        put("standard",      StandardTokenizerFactory.class);
+        put("thai",          ThaiTokenizerFactory.class);
+        put("uax29urlemail", UAX29URLEmailTokenizerFactory.class);
+        put("whitespace",    WhitespaceTokenizerFactory.class);
+        
+        // this one "seems to mess up offsets". probably shouldn't be a tokenizer...
+        put("wikipedia",     Void.class);
+    }});
+    
+    static final Map<String,Class<?>> KNOWN_TOKENFILTERS = Collections.unmodifiableMap(new HashMap<String,Class<?>>() {{
```

Comment:
I just moved this code but I don't like these anonymous subclasses of HashMap either. Will fix.